src/main/transformers.js

```diff
@@ -199,3 +200,10 @@ export function getBucketRegionTransformer() {
 export function getBucketNotificationTransformer() {
   return getConcater(xmlParsers.parseBucketNotification)
 }
+
+// Parses GET/SET BucketPolicy response.
+export function getBucketPolicyTransformer(bucketName, objectPrefix) {
+  return getConcater(response => {
+    return parseBucketPolicy(response, bucketName, objectPrefix)
+  })
+}
```

Comment:
do not pass bucketName and objectName here, instead make it return entire json object.
Let the higher layer ie getBucketPolicy deal with converting policy object to BUCKET_POLICY_READ_ONLY etc

so this can be:
```
export getBucketPolicyTransformer = (bucketName, objectPrefix) =>
    getConcater(response => JSON.parse(response))
```